### PR TITLE
SCALA-49: Introduction to Optics with Monocole

### PR DESCRIPTION
**Pull request by barbasa**

### CODE REVIEW

**Comment by barbasa** [comment]
Hi @JonCook , no problem. Amended the naming convention.

File: scala-libraries/src/test/scala/com/baeldung/scala/monocle/OpticsExamplesUnitTest.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import org.scalatest._
+import org.scalatest.Assertions._
+import scala.util.{Success, Failure}
+import OpticsExamples._
+
+class OpticsExamplesUnitTest extends FlatSpec with Matchers {
+
+  "OpticExamples" should "update stock without Lenses" in {
+
```
[the remark this answers is not in the document]
Comment (barbasa) [reply scala-libraries/src/test/scala/com/baeldung/scala/monocle/OpticsExamplesUnitTest.scala:11]: done

File: scala-libraries/src/test/scala/com/baeldung/scala/monocle/OpticsExamplesUnitTest.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    val item = Item("123sku", 35L, currentStock, NoDiscount())
+    val cart = Cart("abc123", item, 1)
+    val user = User("joe", cart)
+
+    val newUser = updateStockWithoutLenses(user)
+    assert(newUser.cart.item.leftInStock == currentStock - 1)
+  }
+
+  it should "update stock with Lenses" in {
+
```
[the remark this answers is not in the document]
Comment (barbasa) [reply scala-libraries/src/test/scala/com/baeldung/scala/monocle/OpticsExamplesUnitTest.scala:22]: done

File: build.sbt [hunk 3]
```diff
@@ -37,3 +37,14 @@ lazy val scala_test = (project in file("scala-test"))
     libraryDependencies += "org.scalamock" %% "scalamock" % "4.4.0" % Test,
     libraryDependencies += "com.novocode" % "junit-interface" % "0.11" % "test"
   )
+
+val monocleVersion = "2.0.0"
```
[the remark this answers is not in the document]
Comment (barbasa) [reply build.sbt]: updated

File: scala-libraries/src/main/scala/com/baeldung/scala/monocle/OpticsExamples.scala [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  case class FixPriceOff(value: Double) extends Discount
+
+  case class User(name: String, cart: Cart)
+  case class Cart(id: String, item: Item, quantity: Int)
+
+  case class Item(
+      sku: String,
+      price: Double,
+      leftInStock: Int,
+      discount: Discount
```
Comment (KevinGilmore) [inline scala-libraries/src/main/scala/com/baeldung/scala/monocle/OpticsExamples.scala:21]: Use two-space indent here.

**Pull request merged**